pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertyDescriptorFactory.java

```diff
@@ -28,30 +30,42 @@
  */
 public abstract class AbstractPropertyDescriptorFactory<T> implements PropertyDescriptorFactory<T> {
 
-    protected static final Map<PropertyDescriptorField, Boolean> CORE_FIELD_TYPES_BY_KEY
-        = CollectionUtil.mapFrom(new PropertyDescriptorField[] {NAME, DESCRIPTION, DEFAULT_VALUE, DELIMITER},
-                                 new Boolean[] {true, true, true, false});
+
+
+
+    static {
+        System.err.println(CORE_FIELD_TYPES_BY_KEY);
```

Comment:
@oowekyala Pushing PropertyDescriptorUtil.ExpectedFieldsBuilder into a stand alone type would seem to break the cycle.  I'll see if I can make a tool to print out an actual static dependency path.